fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexingHeartbeatTest.java

```diff
@@ -0,0 +1,357 @@
+                assertTrue(e.getMessage().contains(testThrowMsg));
+            }
+        }
+
+        for (Index index : indexes) {
+            try (OnlineIndexer indexer = newIndexerBuilder(index).build()) {
+                Assertions.assertThat(indexer.getIndexingHeartbeats(0)).isEmpty();
+            }
+        }
+    }
+
+    @Test
+    void testMutualIndexersHeartbeatsClearAfterBuild() throws InterruptedException {
+        // Check heartbeats count during mutual indexing
+        List<Index> indexes = new ArrayList<>();
+        indexes.add(new Index("indexA", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+        indexes.add(new Index("indexC", field("num_value_unique"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+        int numRecords = 77;
+        populateData(numRecords);
+        int boundarySize = 5;
+        final List<Tuple> boundariesList = getBoundariesList(numRecords, boundarySize);
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = allIndexesHook(indexes);
+        openSimpleMetaData(hook);
+        disableAll(indexes);
+
+        Semaphore pauseSemaphore = new Semaphore(1);
+        Semaphore startSemaphore = new Semaphore(1);
+        final AtomicInteger count = new AtomicInteger(0);
+        pauseSemaphore.acquire();
+        startSemaphore.acquire();
+        AtomicReference<Map<UUID, IndexBuildProto.IndexBuildHeartbeat>> heartbeats = new AtomicReference<>();
+        IntStream.rangeClosed(1, 4).parallel().forEach(i -> {
+            if (i == 4) {
+                try {
+                    startSemaphore.acquire();
+                    Thread.sleep(100);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    throw new RuntimeException(e);
+                }
+                try (OnlineIndexer indexer = newIndexerBuilder(indexes).build()) {
+                    heartbeats.set(indexer.getIndexingHeartbeats(0));
+                }
+                startSemaphore.release();
+                pauseSemaphore.release();
+            } else {
+                AtomicInteger counter = new AtomicInteger(0);
+                try (OnlineIndexer indexer = newIndexerBuilder(indexes)
+                        .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                                .setMutualIndexingBoundaries(boundariesList))
+                        .setConfigLoader(old -> {
+                            if (counter.incrementAndGet() > 0) {
+                                if (count.incrementAndGet() == 2) {
+                                    startSemaphore.release();
+                                }
+                                try {
+                                    pauseSemaphore.acquire();
+                                } catch (InterruptedException e) {
+                                    throw new RuntimeException(e);
```

Comment:
This should re-interrupt the current thread. 
Probably also could use `Assertions.assertDoesNotThrow`

Reply:
v